pkg/components/prometheus-operator/component_conversion_test.go

```diff
@@ -0,0 +1,280 @@
+package prometheus
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"gopkg.in/yaml.v2"
+
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+func renderManifests(t *testing.T, configHCL string) map[string]string {
+	name := "prometheus-operator"
+
+	component := newComponent()
+
+	body, diagnostics := util.GetComponentBody(configHCL, name)
+	if diagnostics != nil {
+		t.Fatalf("Getting component body: %v", diagnostics)
+	}
+
+	diagnostics = component.LoadConfig(body, &hcl.EvalContext{})
+	if diagnostics.HasErrors() {
+		t.Fatalf("Valid config should not return an error, got: %v", diagnostics)
+	}
+
+	ret, err := component.RenderManifests()
+	if err != nil {
+		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
+	}
+
+	return ret
+}
+
+//nolint:funlen
+func TestConversion(t *testing.T) {
+	testCases := []struct {
+		Name                   string
+		InputConfig            string
+		ExpectedConfigFileName string
+		ExpectedConfig         string
+	}{
+		{
+			Name: "use external_url param",
+			InputConfig: `
+component "prometheus-operator" {
+  prometheus {
+    external_url = "https://prometheus.externalurl.net"
+  }
+}
+`,
+			ExpectedConfigFileName: "prometheus-operator/templates/prometheus/prometheus.yaml",
+			ExpectedConfig: `
+# Source: prometheus-operator/templates/prometheus/prometheus.yaml
```

Comment:
Do we really need those content-based tests? I'm really afraid that they will make updating components a huge pain.

Reply:
Changed it, PTAL.